test/javascripts/components/global-filter/filtered-tags-chooser-test.js

```diff
@@ -0,0 +1,58 @@
+import { visit } from "@ember/test-helpers";
+import { acceptance, query } from "discourse/tests/helpers/qunit-helpers";
+import { test } from "qunit";
+import selectKit from "discourse/tests/helpers/select-kit-helper";
+
+acceptance("Discourse Global Filter - Filtered Tags Chooser", function (needs) {
+  needs.settings({
+    discourse_global_filter_enabled: true,
+    global_filters: "support",
+    tagging_enabled: true,
+  });
+
+  needs.pretender((server, helper) => {
+    server.get("/tag/support/notifications", () =>
+      helper.response({
+        tag_notification: { id: "support", notification_level: 2 },
+      })
+    );
+
+    server.get("/tag/support/l/latest.json", () => {
+      return helper.response({
+        users: [],
+        primary_groups: [],
+        topic_list: {
+          can_create_topic: true,
+          draft: null,
+          draft_key: "new_topic",
+          draft_sequence: 1,
+          per_page: 30,
+          tags: [],
+          topics: [],
+        },
+      });
+    });
+
+    server.put("/global_filter/filter_tags/support/assign.json", () => {
+      return helper.response({ success: true });
+    });
+  });
+
+  test("hides the selected global filter tag from the tag-drop chooser", async function (assert) {
+    await visit("/tag/support");
+
+    assert.strictEqual(
+      query(".filtered-tag-drop .tag-drop .name").innerText,
+      "all tags",
+      "does not display the selected global filter"
+    );
+
+    const tags = selectKit(".filtered-tag-drop .tag-drop");
+    await tags.expand();
+
+    assert.notOk(
+      tags.rowByValue("support").exists(),
+      "global filter is not displayed as selected"
+    );
+  });
+});
```

Comment:
Since support is a GTF, it should be omitted from the tag-drop element everywhere, right? Can we add a second test case for that, i.e. if you're on another route, ensure that tag drop does not include GTF.